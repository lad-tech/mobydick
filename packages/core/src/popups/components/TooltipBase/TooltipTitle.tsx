import {FC} from 'react';

import {ITitlePopup} from '../PopupBase';
import useStyles from '../../../styles/hooks/useStyles';
import {TypographyLegacy} from '../../../typography/components/TypographyLegacy/TypographyLegacy';

import stylesCreate from './stylesCreate';

const TooltipTitle: FC<ITitlePopup> = props => {
  const [styles] = useStyles(stylesCreate);
  const {title, titleStyles, titleFont} = props;

  return (
    <TypographyLegacy
      font={titleFont ? titleFont : 'SemiBold-Contrast-XS'}
      style={[styles.title, titleStyles]}>
      {title}
    </TypographyLegacy>
  );
};

export default TooltipTitle;
