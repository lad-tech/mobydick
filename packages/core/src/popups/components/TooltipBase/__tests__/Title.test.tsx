import {render} from '@testing-library/react-native';

import TooltipTitle from '../TooltipTitle';

describe('@lad-tech/mobydick-core/TooltipBase/Title', () => {
  it('should renders correctly font', () => {
    const {toJSON} = render(
      <TooltipTitle
        title={'title'}
        titleFont={'Regular-Primary-XS'}
        titleStyles={{width: 100}}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly', () => {
    const {toJSON} = render(<TooltipTitle title={'title'} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
