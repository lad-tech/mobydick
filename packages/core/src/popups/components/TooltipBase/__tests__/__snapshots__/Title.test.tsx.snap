// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lad-tech/mobydick-core/TooltipBase/Title should renders correctly 1`] = `
<Text
  allowFontScaling={true}
  style={
    [
      {
        "color": "#FFFFFF",
        "fontFamily": "Inter-SemiBold",
        "fontSize": 14,
        "fontWeight": "600",
        "lineHeight": 18,
        "minHeight": 18,
      },
      [
        {
          "zIndex": 1,
        },
        undefined,
      ],
    ]
  }
>
  title
</Text>
`;

exports[`@lad-tech/mobydick-core/TooltipBase/Title should renders correctly font 1`] = `
<Text
  allowFontScaling={true}
  style={
    [
      {
        "color": "#20242D",
        "fontFamily": "Inter-Regular",
        "fontSize": 12,
        "fontWeight": "400",
        "lineHeight": 16,
        "minHeight": 16,
      },
      [
        {
          "zIndex": 1,
        },
        {
          "width": 100,
        },
      ],
    ]
  }
>
  title
</Text>
`;
